src/Diagnostics.DataProviders/DataProviders/SupportObserverDataProvider.cs

```diff
@@ -226,16 +228,48 @@ public override async Task<dynamic> GetHostingEnvironmentPostBody(string hosting
             return hostingEnvironmentPostBody;
         }
 
+        public override async Task<DataTable> ExecuteSqlQueryAsync(string cloudServiceName, string query)
+        {
+            if (!query.StartsWith("\""))
+            {
+                // BUG: Passing as JSON requires query to be wrapped in quotes, fix API expected content type
+                query = $"\"{query}\"";
+            }
+
+            var request = new HttpRequestMessage(HttpMethod.Post, $"/api/service/{cloudServiceName}/invokesql?api-version=2.0")
+            {
+                Content = new StringContent(query, Encoding.Default, "application/json")
+            };
+
+            var response = await SendObserverRequestAsync(request);
+            response.EnsureSuccessStatusCode();
+
+            var result = await response.Content.ReadAsStringAsync();
+
+            DataTable datatable;
+            try
+            {
```

Comment:
should we put try catch around the API call itself in case server returns errors like 5xx?

Reply:
The `response.EnsureSuccessStatusCode()` call handles the API returning non-2xx responses.
This try/catch just handles unexpected data being returned even though a 2xx response was received, such as a message string.

Reply:
I see. Ok